signum/signum.md created (#1122)
Codecademy/docs: content/java/concepts/math-methods/terms/signum/signum.md

---
Title: '.signum()'

> Description: 'Used to determine whether a number is positive, negative, zero or NaN'

Reviewer: ```suggestion
Description: 'Used to determine whether a number is positive, negative, zero, or NaN'
```

> Description: 'Used to determine whether a number is positive, negative, zero, or NaN'

Reviewer: ```suggestion
Description: 'Used to determine whether a number is positive, negative, zero, or NaN.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Arithmetic'
  - 'Numbers'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`Math.signum()`** determines whether a number is positive, negative, zero or NaN.

Reviewer: ```suggestion
The **`Math.signum()`** determines whether a number is positive, negative, zero, or NaN.
```

> The **`Math.signum()`** determines whether a number is positive, negative, zero, or NaN.

Reviewer 2: ```suggestion
The **`Math.signum()`** method determines whether a number is positive, negative, zero, or `NaN`.
```

## Syntax

```pseudo
Math.signum(num)
```

> The `.signum()` method takes a double `num` parameter and returns a double.

Reviewer: ```suggestion
The `.signum()` method takes a `double` `num` parameter and returns a `double`.
```

> The `.signum()` method takes a `double` `num` parameter and returns a `double`.

Reviewer 2: ```suggestion
The `.signum()` method takes a `num` parameter and returns a `double` value that indicates whether `num` is positive, negative, zero, or `NaN`.
```

> Some edge cases for the `.signum()` method include the following:
> 
> - If the value of `num` is `NaN`, `NaN` is returned.

Reviewer 2: ```suggestion
Some cases to consider include the following:

- If `num` is positive, a positive non-zero value is returned.
- If `num` is negative, a negative non-zero value is returned.
- If `num` is zero, zero is returned.
- If the value of `num` is `NaN`, `NaN` is returned.
```

## Example

> The following example demonstrates the `.signum()` method:

Reviewer 2: ```suggestion
The following example demonstrates the `.signum()` method and its many use cases:
```

```java

> public class Test {

Reviewer 2: ```suggestion
// Test.java
public class Test {
```

  public static void main(String[] args) {

> 
>     // Returns 1.0 when argument is positive
>     System.out.println(Math.signum(8.75));
>     // Returns -1.0 when argument is negative
>     System.out.println(Math.signum(-3.5));
>     // Returns 0 when argument is zero.
>     System.out.println(Math.signum(0));
> 
>     // Edge cases
>     System.out.println(Math.signum(Double.NaN));

Reviewer 2: ```suggestion
    System.out.println(Math.signum(8.75));
    System.out.println(Math.signum(-3.5));
    System.out.println(Math.signum(0));
    System.out.println(Math.signum(Double.NaN));
```
